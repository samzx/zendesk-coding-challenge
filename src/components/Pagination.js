import React from 'react';

export const itemsPerPage = 25;

class Pagination extends React.Component {

    createPagination = (count) => {
        const numPages = Math.floor(count / itemsPerPage);
        let arr = []
        for(let i=1; i<=numPages; i++) {
            arr.push(i);
        }
        return (
            arr.map((page) => 
                <div
                    key={`page-${page}`}
                    onClick={() => {
                        this.props.fetchListings(page);
                    }}
                >
                    {
                        this.props.currentPage == page ?
                        <b>{page}</b> :
                        page
                    }
                </div>
            )
        );
    }
    
    render() {
        return(
            <div className="pagination" style={{display: 'flex', margin: 'auto'}}>
                {
                    this.props.data &&
                    this.createPagination(this.props.data.count)
                }
            </div>
        );
    }
}

export default Pagination;