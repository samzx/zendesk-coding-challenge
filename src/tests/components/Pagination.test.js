import React from "react";
import { shallow } from "enzyme";
import Pagination from "../../components/Pagination";

test("should render Pagination correctly", () => {
  const wrapper = shallow(<Pagination />);
  expect(wrapper).toMatchSnapshot();
});
