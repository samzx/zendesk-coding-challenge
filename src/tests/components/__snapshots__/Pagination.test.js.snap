// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Pagination correctly 1`] = `
<div
  className="pagination"
/>
`;
